open_seq2seq/parts/tacotron/tacotron_decoder.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2018 NVIDIA Corporation
+
+from __future__ import absolute_import, division, print_function
+from __future__ import unicode_literals
+
+import collections
+
+from tensorflow.contrib.seq2seq.python.ops import decoder
+from tensorflow.contrib.seq2seq.python.ops import helper as helper_py
+from tensorflow.python.framework import ops
+from tensorflow.python.framework import tensor_shape
+from tensorflow.python.layers import base as layers_base
+from tensorflow.python.ops import rnn_cell_impl
+from tensorflow.python.util import nest
+from tensorflow.python.ops import array_ops
+
+
+class BasicDecoderOutput(
+    collections.namedtuple("BasicDecoderOutput", ("rnn_output", "target_output", "sample_id"))):
+  pass
+
+
+class TacotronDecoder(decoder.Decoder):
+  """Basic sampling decoder."""
+
+  def __init__(self, decoder_cell, attention_cell, helper,
+              initial_decoder_state, initial_attention_state, 
+              attention_type, spec_layer, target_layer,
+              use_prenet_output=True):
+    """Initialize BasicDecoder.
+    Args:
```

Comment:
You need explicit empty lines before Args and Returns to make sure it's rendered correctly in the docs